mobile/v1/snowflake/sql-runner/sql/standard/01-base/01-main/00-setup-base.sql

```diff
@@ -0,0 +1,271 @@
+CREATE TABLE IF NOT EXISTS {{.scratch_schema}}.{{.model}}_metadata_run_id{{.entropy}} (
+  run_id TIMESTAMP_NTZ
+);
+
+-- When base runs, it's always the first module. So it's safe to just truncate here.
+TRUNCATE {{.scratch_schema}}.{{.model}}_metadata_run_id{{.entropy}};
+
+INSERT INTO {{.scratch_schema}}.{{.model}}_metadata_run_id{{.entropy}} (
+  SELECT
+    CURRENT_TIMESTAMP::TIMESTAMP_NTZ
+);
+
+-- Permanent metadata table
+CREATE TABLE IF NOT EXISTS {{.output_schema}}.datamodel_metadata{{.entropy}} (
+  run_id                     TIMESTAMP_NTZ,
+  model_version              VARCHAR(64),
+  model                      VARCHAR(64),
+  module                     VARCHAR(64),
+  run_start_tstamp           TIMESTAMP_NTZ,
+  run_end_tstamp             TIMESTAMP_NTZ,
+  rows_this_run              INTEGER,
+  distinct_key               VARCHAR(64),
+  distinct_key_count         INTEGER,
+  time_key                   VARCHAR(64),
+  min_time_key               TIMESTAMP_NTZ,
+  max_time_key               TIMESTAMP_NTZ,
+  duplicate_rows_removed     INTEGER,
+  distinct_keys_removed      INTEGER
+);
+
+-- Setup temp metadata tables for this run
+CREATE OR REPLACE TABLE {{.scratch_schema}}.{{.model}}_base_metadata_this_run{{.entropy}} (
+  id                         VARCHAR(64),
+  run_id                     TIMESTAMP_NTZ,
+  model_version              VARCHAR(64),
+  model                      VARCHAR(64),
+  module                     VARCHAR(64),
+  run_start_tstamp           TIMESTAMP_NTZ,
+  run_end_tstamp             TIMESTAMP_NTZ,
+  rows_this_run              INTEGER,
+  distinct_key               VARCHAR(64),
+  distinct_key_count         INTEGER,
+  time_key                   VARCHAR(64),
+  min_time_key               TIMESTAMP_NTZ,
+  max_time_key               TIMESTAMP_NTZ,
+  duplicate_rows_removed     INTEGER,
+  distinct_keys_removed      INTEGER
+);
+
+INSERT INTO {{.scratch_schema}}.{{.model}}_base_metadata_this_run{{.entropy}} (
+  SELECT
+    'run',
+    run_id,
+    '{{.model_version}}',
+    '{{.model}}',
+    'base',
+    CURRENT_TIMESTAMP::TIMESTAMP_NTZ,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL
+  FROM {{.scratch_schema}}.{{.model}}_metadata_run_id{{.entropy}}
+);
+
+
+-- Setup manifests
+CREATE TABLE IF NOT EXISTS {{.output_schema}}.{{.model}}_base_event_id_manifest{{.entropy}}
+AS (
+  SELECT
+    'seed'::VARCHAR(36) AS event_id,
+    '{{.start_date}}'::TIMESTAMP_NTZ AS collector_tstamp
+);
+
+CREATE TABLE IF NOT EXISTS {{.output_schema}}.{{.model}}_base_session_id_manifest{{.entropy}}
+AS (
+  SELECT
+    'seed'::VARCHAR(36) AS session_id,
+    '{{.start_date}}'::TIMESTAMP_NTZ AS min_tstamp
+);
+
+
+-- Create staged table
+-- Not defining mobile_events_staged as table likely to evolve. Suggest we remove web def below in future releases of SF model.
+{{if eq .model "web"}}
```

Comment:
Why do we have an explicit table definition for web but not mobile?

Reply:
Ultimately the staged table is going to be mutated by the `commit_table` function as automigrate = true so it seemed redundant to define the staging table. Personally I think it makes sense to remove the web definition. Thoughts?

Reply:
On one hand, it can be useful to have the base fields defined somewhere for dev accessibility purposes. On the other, you're right that it is indeed redundant to do so. I'm happy with either call, I just think it should be consistent. :)

Reply:
That is a fair point but I think it is slightly confusing to have this strict definition and then happily mutate it later on. I am going to remove.